### ISSUE TITLE
raw_value() on a leaf-list incorrect

### ISSUE THREAD

**Issue by kwatsen** [CZ-NIC/yangson#70]
Adding this code to tests/test_model.py note the import):
```
from yangson.instance import RootNode

def test_rpcA_output(data_model):
    output_obj = {
        "testb:output" : {
            "llistC" : [True, False, True]
        }
    }
    print("\n")
    sn = data_model.get_schema_node("/testb:rpcA")
    cooked = sn.from_raw(output_obj, allow_nodata=True)
    print("cooked = " + str(cooked))
    inst= RootNode(cooked, sn, data_model.schema_data, cooked.timestamp)
    rv = inst.raw_value()
    print("rv = " + str(rv))
    assert(rv == output_obj)
```

And running this command:
```
pytest -sxv -k test_rpcA_output
```

Produces this output (note that the "cooked" value is correct):
```
cooked = {'testb:output': {'llistC': [True, False, True]}}
rv = {'testb:output': {'llistC': [True, True]}}

E       AssertionError: assert {'testb:outpu...[True, True]}} == {'testb:outpu...False, True]}}
E         Differing items:
E         {'testb:output': {'llistC': [True, True]}} != {'testb:output': {'llistC': [True, False, True]}}
E         Full diff:
E         - {'testb:output': {'llistC': [True, False, True]}}
E         ?                                   -------
E         + {'testb:output': {'llistC': [True, True]}}
```

**Comment by llhotka** [comment]
I am actually getting something different:
```
E           yangson.exceptions.NonexistentInstance: [/] member 'output'
```
This sort of makes sense because `rpcA` and `output` schema nodes are in the same namespace, so the `output` instance is expected to have no namespace.

The problem is that `RootNode` was intended for the dummy root of a normal instance tree whose schema node is SchemaTreeNode that has no name and no namespace.

**Comment by llhotka** [comment]
I get the same result as you with this modified test function:
```python
def test_rpcA_output(data_model):
    output_obj = {
        "testb:output" : {
            "llistC" : [True, False, True]
        }
    }
    print("\n")
    sn = data_model.get_schema_node("/testb:rpcA")
    cooked = sn.from_raw(output_obj, allow_nodata=True)
    print("cooked = " + str(cooked))
    sn.name = sn.ns = None
    inst= RootNode(cooked, sn, data_model.schema_data, cooked.timestamp)
    rv = inst.raw_value()
    print("rv = " + str(rv))
    assert(rv == output_obj)
```
Note the line `sn.name = sn.ns = None`

**Comment by kwatsen** [comment]
FWIW, the code "works" is the leaf-list is changed to `[True, True, True]`.

It seems that the `False` value is being stripped?

**Comment by kwatsen** [comment]
BTW, I don't like the "allow_nodata" parameter that was added.  

It's not documented and, even after looking at how the code acts on it, it still doesn't make sense to me.   

I'm my view, the code should be more auto-sensing, based on where in the schema tree it is...

**Comment by kwatsen** [comment]
Toggling "allow_nodata" (amongst possible other things) effects whether the top-level node is prefixed.

- `True` generates `cooked = {'testb:output': {'llistC': [True, False, True]}}`
- `False` generates: `cooked = {'output': {'llistC': [True, False, True]}}`


This makes me wonder what the **correct** cooked value is...please say if the cooked value should or should not be prefixed here.

_Looking at the `iname()` method, I think that cooked values in general SHOULD be prefixed when needed...and (answering my previous question) the `output` node SHOULD be prefixed._

**Comment by llhotka** [comment]
Here is the line that causes the problem:
```
(yangson) $ git blame -L 475 yangson/instance.py | head -1
d6e1203c (Henning Rogge   2020-05-07 08:19:17 +0200  475)                 if add1 and add2 and member_value:
```
If `member_value` is a boolean and happens to be `False`, then it is not added to the raw value under construction. I wonder if it should be instead:
```
if add1 and add2 and member_value is None:
```
@HRogge can you please fix it?

**Comment by HRogge** [comment]
Fixed in PR #73

**Comment by llhotka** [comment]
I complemented PR #73 in order to resolve this issue completely, i.e. including the namespace of `output` (see my first comment above). I also added Kent's example to the test suite. The result is commit d7cbfdaa12b877de644bf27449ae8757c832e1f7.

I had to make two changes that are IMO OK but may potentially break old code:
* Namespace of `RootNode` is now `None`
* `InputNode` and `OutputNode` are now subclasses of `InternalNode` rather than `SchemaTreeNode` (along with `DataNode`)
